src/owr/OptimisticWithdrawalRecipientPectra.sol

```diff
@@ -181,6 +194,19 @@ contract OptimisticWithdrawalRecipientPectra is Clone, OwnableRoles {
   /*     emit ReceiveETH(msg.value); */
   /* } */
 
+  function requestConsolidation(bytes memory data) external payable onlyOwnerOrRoles(CONSOLIDATION_WITHDRAWAL_ROLE) {
+    // ;; Input data has the following layout:
+    // ;;
+    // ;;  +--------+--------+
+    // ;;  | source | target |
+    // ;;  +--------+--------+
+    // ;;      48       48
+
+    (bool ret,) = pectraConsolidationAddress().call{value: msg.value}(data);
+    if (!ret) revert InvalidConsolidation_Failed();
+    emit ConsolidationRequested(msg.sender);
+  }
+
   /// Requests principal withdrawal
   function requestPrincipalWithdrawal(bytes memory data) external payable onlyOwnerOrRoles(PRINCIPAL_WITHDRAWAL_ROLE) {
```

Comment:
Request withdrawal should consider the actual data types and not bytes data. We can convert to bytes in solidity

Reply:
Refactored them to receive `pubkey` and `amount`